### DISCUSSION TITLE
catching custom events with hypens in the event name eg event-name

### DISCUSSION THREAD

**Discussion by Jero9999** [lit/lit#4236] [Q&A] [answered]
I am writing my own component(s) and I am in need of handing a custom event in a child component. Normally this would work
`html"<child-component @customEvent=${myEventHandler}></child-component>"`

But the event in question has a hyphen in it (eg my-custom-event). So that changes the template to :
`html"<child-component @my-custom-event=${myEventHandler}></child-component>"

and that never seems to trigger the event handler. 
I have looked around for any guidance on what syntax is legal and can't see anything. If it is in the lit docs, then I cannot find it.

More generally, I have looked for guidance on what naming conventions there are for synthetic events and I can't find any there either.

Can someone help? **should** `@my-event-${handler}` work? Am I just doing something wrong? Or are supported names more restrictive that this?

**Comment by tbroyer** [comment, marked as answer]
This is supposed to work with `@my-event=${myEventHandler}`, there are a few tests in Lit that use kebab-case like that, e.g. https://github.com/lit/lit/blob/47c6c90b2db5d851f7fd11fe50d412c4df37b301/packages/lit-html/src/test/lit-html_test.ts#L1403-L1430
